src/main/java/org/openstreetmap/atlas/geography/atlas/items/complex/bignode/BigNodeFinder.java

```diff
@@ -440,6 +461,55 @@ protected boolean hasJunctionEdgeTags(final Edge edge)
         return true;
     }
 
+    // Check if merged junction candidate is a very long straight route which is normally not a
+    // valid junction route
+    protected boolean isStraightLongRoute(final Route mergedCandidate)
+    {
+        // if a junction route is merged route, it must have equal to or more than two junction
+        // edges
+        if (mergedCandidate.size() < 2)
+        {
+            return false;
+        }
+        // if merged candidate is longer than threshold, we will check if all junctions edges are
+        // from the same long and straight road
+        if (mergedCandidate.length().isGreaterThanOrEqualTo(this.longJunctionRouteLength))
+        {
+            final Iterator<Edge> iterator = mergedCandidate.iterator();
+            Edge currentEdge = iterator.next();
+            while (iterator.hasNext())
+            {
+                final Edge nextEdge = iterator.next();
+                final Optional<Heading> currentHeading = currentEdge.overallHeading();
+                final Optional<Heading> nextHeading = nextEdge.overallHeading();
+                // if the angle of any two edges are large than 60 degree, we think they are valid
+                // junction route
+                if (currentHeading.isPresent() && nextHeading.isPresent()
+                        && currentHeading.get().difference(nextHeading.get()).asPositiveAngle()
+                                .isGreaterThanOrEqualTo(this.nonStraightJunctionEdgesAngle))
+                {
+                    return false;
+                }
+                // if there are any two edges has different name, we think they are valid
+                if (currentEdge.getTag(NameTag.KEY).isPresent()
+                        && nextEdge.getTag(NameTag.KEY).isPresent()
+                        && !currentEdge.getTag(NameTag.KEY).get()
+                                .equalsIgnoreCase(nextEdge.getTag(NameTag.KEY).get()))
+                {
+                    return false;
+                }
+                currentEdge = nextEdge;
+            }
+            logger.info("Invalid Merged Candidate Route length is {} with WKT {}",
+                    mergedCandidate.length(), mergedCandidate.asPolyLine().toWkt());
```

Comment:
```suggestion
        if (logger.isInfoEnabled())
        {
            logger.info("Invalid Merged Candidate Route length is {} with WKT {}",
                    mergedCandidate.length(), mergedCandidate.asPolyLine().toWkt());
        }
```

Reply:
This might help performance. Prevents computations from occurring unnecessarily when the log level is set higher.